TLM/TLM/UI/IncompatibleModsPanel.cs

```diff
@@ -139,27 +151,43 @@ public class IncompatibleModsPanel : UIPanel {
         }
 
         private UIPanel CreateEntry(ref UIScrollablePanel parent, string name, ulong steamId) {
+            string caption = steamId == LOCAL_TMPE ? "Delete" : "Unsubscribe";
+
             UIPanel panel = parent.AddUIComponent<UIPanel>();
             panel.size = new Vector2(560, 50);
             panel.backgroundSprite = "ContentManagerItemBackground";
+
             UILabel label = panel.AddUIComponent<UILabel>();
             label.text = name;
             label.textAlignment = UIHorizontalAlignment.Left;
             label.relativePosition = new Vector2(10, 15);
-            CreateButton(panel, "Unsubscribe", (int) panel.width - 170, 10, delegate(UIComponent component, UIMouseEventParameter param) { UnsubscribeClick(component, param, steamId); });
+
+            CreateButton(panel, caption, (int)panel.width - 170, 10, delegate (UIComponent component, UIMouseEventParameter param) { UnsubscribeClick(component, param, steamId); });
+
             return panel;
         }
 
         private void UnsubscribeClick(UIComponent component, UIMouseEventParameter eventparam, ulong steamId) {
-            Log.Info("Trying to unsubscribe workshop item " + steamId);
+
+            bool success = false;
+
             component.isEnabled = false;
-            if (PlatformService.workshop.Unsubscribe(new PublishedFileId(steamId))) {
+
+            if (steamId == LOCAL_TMPE) {
+                Log.Info("Trying to delete local build of TM:PE");
+                success = DeleteLocalTMPE();
+            } else {
+                Log.Info("Trying to unsubscribe workshop item " + steamId);
+                success = PlatformService.workshop.Unsubscribe(new PublishedFileId(steamId));
+            }
+
+            if (success) {
                 IncompatibleMods.Remove(steamId);
                 component.parent.Disable();
                 component.isVisible = false;
-                Log.Info("Workshop item " + steamId + " unsubscribed");
+                Log.Info("Successfully removed incompatbile mod");
             } else {
-                Log.Warning("Failed unsubscribing workshop item " + steamId);
+                Log.Warning("Failed to remove incompatible mod");
```

Comment:
I have got two local builds but only one were listed ((this build and second from harmony) + sub to LABS from workshop and yours [Stable] => 4 but listed 2 only 😃 ) then after click on remove button, harmony version remains on list of mods to enable and log were flooded with `DirectoryNotFoundException` just after I hit enable :smile:
I will test this case later...

BTW What if detector runs from local build and we want to remove it to leave only that one subscribed from workshop?

Reply:
Ah, yeah, the way it works currently it expects there will only ever be a single local version. I can fix that when implementing the plugin helper class.

Reply:
Fixed.